File: 백준/Silver/17219. 비밀번호 찾기/비밀번호 찾기.py
import sys

input = sys.stdin.readline

N,M = map(int,input().split())
obj = {}

for _ in range(N) : 
  site,ps = input().rstrip().split()

  if site not in obj :
    obj[site] = ps

for _ in range(M) :
  FS = input().rstrip()

  print(obj[FS])

File: 백준/Silver/17219. 비밀번호 찾기/비밀번호 찾기.js
const readLine = require('fs').readFileSync('dev/stdin').toString().trim();
const [N, ...input] = readLine.split('\n');
const [K, M] = N.split(' ').map((num) => Number(num));
const map = new Map();
const answer = [];

for (const x of input.splice(0, K)) {
  const [siteUrl, password] = x.split(' ');

  map.set(siteUrl, password);
}

for (const siteUrl of input) {
  answer.push(map.get(siteUrl));
}

console.log(answer.join('\n'));